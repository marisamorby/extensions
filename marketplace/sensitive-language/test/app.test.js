'use strict';

const React = require('react');
const renderer = require('react-test-renderer');

const {mockComponent} = require('./mock-component.js');

jest.mock('../src/unsupported-language.js', () => ({
  UnsupportedLanguage: mockComponent('unsupported-language')
}));
jest.mock('../src/language-checker.js', () => ({
  LanguageChecker: mockComponent('language-checker')
}));

const {App} = require('../src/app.js');

test('App: bails out for non-english languages', function () {
  const extension = {
    locales: {
      default: 'de-DE',
      names: {
        'de-DE': 'German'
      }
    },
    window: {
      startAutoResizer() {}
    }
  };

  const tree = renderer
    .create(<App extension={extension} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('App: works for English languages', function () {
  const extension = {
    locales: {
      default: 'en-US',
      names: {
        'en-US': 'English'
      }
    },
    window: {
      startAutoResizer() {}
    },
    contentType: {
      fields: [
        {
          id: '123',
          type: 'Symbol'
        }
      ]
    },
    entry: {
      sys: {
        id: 'ABC'
      }
    }
  };

  const tree = renderer
    .create(<App extension={extension} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('App: filters out non-text fields', function () {
  const extension = {
    locales: {
      default: 'en-US',
      names: {
        'en-US': 'English'
      }
    },
    window: {
      startAutoResizer() {}
    },
    contentType: {
      fields: [
        {
          id: '123',
          type: 'Symbol'
        },
        {
          id: '456',
          type: 'Number'
        },
        {
          id: '789',
          type: 'Text'
        },
        {
          id: '012',
          type: 'RichText'
        },
        {
          id: '345',
          type: 'Boolean'
        }
      ]
    },
    entry: {
      sys: {
        id: 'ABC'
      }
    }
  };

  const tree = renderer
    .create(<App extension={extension} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
