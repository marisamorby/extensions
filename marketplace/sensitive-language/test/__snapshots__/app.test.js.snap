// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App: bails out for non-english languages 1`] = `
<div
  className="unsupported-language"
  localeCode="de-DE"
  localeName="German"
/>
`;

exports[`App: filters out non-text fields 1`] = `
<div
  className="language-checker"
  entry={
    Object {
      "sys": Object {
        "id": "ABC",
      },
    }
  }
  fieldsToCheck={
    Array [
      Object {
        "id": "123",
        "type": "Symbol",
      },
      Object {
        "id": "789",
        "type": "Text",
      },
      Object {
        "id": "012",
        "type": "RichText",
      },
    ]
  }
/>
`;

exports[`App: works for English languages 1`] = `
<div
  className="language-checker"
  entry={
    Object {
      "sys": Object {
        "id": "ABC",
      },
    }
  }
  fieldsToCheck={
    Array [
      Object {
        "id": "123",
        "type": "Symbol",
      },
    ]
  }
/>
`;
